src/adhocracy_core/docs/workflows.rst

```diff
@@ -0,0 +1,119 @@
+    >>> from adhocracy_core.resources.sample_proposal import IProposal
+    >>> app_god = getfixture('app_god')
+    >>> app_god.base_path = '/adhocracy'
+
+Lets create some content::
+
+    >>> data = {'adhocracy_core.sheets.name.IName': {'name': 'proposals'}}
+    >>> resp = app_god.post_resource('/', IBasicPool, data)
+    >>> data = {'adhocracy_core.sheets.name.IName': {'name': 'proposal_item'}}
+    >>> resp = app_god.post_resource('/proposals', IProposal, data)
+
+
+Workflows
+---------
+
+Workflows are fine state machines Assigned to a Resource.
+States can set the local permissions.
+States can have metadata (title, date,...).
+State transitions can have a callable to excecute arbitrary tasks.
+TODO nice Introduction.
+
+The MetaAPI gives us the states and transitions metadata for each workflow::
+
+    >>> resp_data = app_god.get('/../meta_api').json
+    >>> worklow = resp_data['worklows']['sampleworkflow']
+
+State metadata contains a human readable title ::
+
+    >>> state = workflow['states']['announced']
+    >>> state['title']
+    'Announced'
+
+a description ::
+
+    >>> state['description']
+    'In this phase you may wait.'
+
+a local ACL (see doc:`authorization`) that is set when entering this state, ::
+
+    >>> state['acl']
+    [('Disallow, 'role:reader', 'view')]
+
+a hint for the frontend if displaying this state in listing should be restricted::
+
+    >>> state['display_only_to_roles']
+    ['Manager']
+
+The order these states shoudl be listet is also set, in addition this
+defines the initial workflow state (the first in the list). ::
+
+    >>> workflow['states_order']
+    ['draft', 'announced', 'published']
+
+Transition metadata determes the possible state flow and can provide a callable to
+excecute arbitrary tasks ::
+
+     >>> transition = workflow['transitions']['to_announced']
+     >>> pprint(transition)
+     {
+      'callback': 'adhocracy_core.workflows.samplecallback',
+      'from_state': 'draft',
+      'to_state': 'announced',
+      'permission': 'review',
+      }
+
+
+Workflow Assigment
+------------------
+
+Resources have a WorkflowAssigenment sheet to assign the wanted workflow::
+
+    >>> resp_data = app_god.get('/proposals/proposal_item').json
+    >>> workflow_data = resp_data['data']['adhocracy_core.sheets.workflow.ISampleWorkflowAssignment']
+    >>> workflow_data['workflow_type']
+    'sampleworkflow'
+
+in addition we can add custom metadata for specific workflow states::
+
+    >>> workflow_data['announced']['start_date']
+    '2015-02-14...,
+    >>> workflow_data['announced']['description]
+    'Soon you can participate...
+
+
+Workflow transition to states
+-----------------------------
+
+The Metadata sheets shows the current workflow state::
+
+    >>> resp_data = app_god.get('/proposals/proposal_item').json
+    >>> metadata_data = resp_data['data']['adhocracy_core.sheets.metadata.IMetadataSheet']
+    >>> workflow_data['workflow_state']
+    'announced'
```

Comment:
> As discussed above I do not yet understand why every resource should have a workflow. And putting workflow related data in the metadata sheet works only in that case.

Workflow could be usefull for differen resource types, resources without it just display None, thats not much overhead.

I proposed to add this information to the metadata sheet because:

* it is some kind of metadata
* we need special permission checks on field level, thats why we need a special sheet, so why not use the metadata sheet thats already there.
* the possible values depend on the workflow and user permission, thats why we need to exposes these with the options request response, the metadata sheet alread has a similar behavior.